### PR TITLE
Readme updates and related files

### PR DESCRIPTION
**Pull request by mohanvive**

### CODE REVIEW

File: shippingservice/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        tracking_id: generateTrackingId(string `${address.street_address}, ${address.city}, ${address.state}`)
+    };
+}
+```
+
+### A Glimpse of comparison
+
+The below figure provides a glimpse of the comparison of the source codes written in the Go and Ballerina languages.
+
+<img src="/images/shipping-service.png">
```
Comment (mohanvive) [inline shippingservice/README.md:39]: I have added a code comparison image. Thoughts? We can do this for all the services if it is something interesting.

Comment (mohanvive) [reply shippingservice/README.md:39]: @sameerajayasoma @xlight05 @dilanSachi @anupama-pathirage

Comment (mohanvive) [reply shippingservice/README.md:39]: Guys, any comments on this? Shall we merge this if you guys are OK with the change.

Comment (xlight05) [reply shippingservice/README.md:39]: Sorry, Missed the notification, Ill look into this now

File: README.md [hunk 2]
```diff
@@ -618,9 +234,17 @@ npm start
...
-![image info](drop-down-services-spans.png)
+![image info](images/drop-down-services-spans.png)
 
 Then click on `Find Traces` and you will be able to view spans of services in the gcp-microservice.
-![image info](tracing-spans.png)
+![image info](images/tracing-spans.png)
 
 For more information about observability in Ballerina, please visit [Observe Ballerina Programs](https://ballerina.io/learn/observe-ballerina-programs/).
+
+# Deviations from [GCP Microservices Demo](https://github.com/GoogleCloudPlatform/microservices-demo)
```
Comment (xlight05) [inline README.md:244]: We need to mention that we broke down frontend service to two services. UI and frontned(BFF service we agreed that day)

File: shippingservice/README.md (hunk 1 above)
Comment (xlight05) [reply shippingservice/README.md:39]: I think, its better than having LOC. shows that the ballerina code is compact and more readable. 
I'm wondering if we should do this to highlight ballerina features as well? For the stuff we have in https://github.com/ballerina-guides/gcp-microservices-demo/blob/main/Highlights.md

File: README.md [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
+The <a href="https://github.com/GoogleCloudPlatform/microservices-demo" target="_blank">online boutique</a> is a cloud-native microservices demo application written by the Google cloud platform. It consists of a 10-tier microservices application. The application is a web-based e-commerce app using which users can browse items, add them to the cart, and purchase them. The microservices in the GCP demo are implemented using multiple programming languages such as Java, C#, Go, Javascript and Python.      
+
+Here, in this demo these microservices are written using [Ballerina language](https://ballerina.io/) to demonstrate the language features, showcase best practices when writing microservices and provide an in-depth understanding of how ballerina services can interact in a real-world scenario. In our implementation, communication between the microservices is handled using gRPC and the frontend is exposed via an HTTP service.
```
Comment (xlight05) [inline README.md:4]: Shall we add deploying in cloud part as well?

File: README.md [hunk 4]
```diff
@@ -618,9 +234,17 @@ npm start
...
 After running the docker container, you can view the tracing information on Jaeger via http://localhost:16686/. You can select the service in the drop-down box as follows.
-![image info](drop-down-services-spans.png)
+![image info](images/drop-down-services-spans.png)
 
 Then click on `Find Traces` and you will be able to view spans of services in the gcp-microservice.
-![image info](tracing-spans.png)
+![image info](images/tracing-spans.png)
 
 For more information about observability in Ballerina, please visit [Observe Ballerina Programs](https://ballerina.io/learn/observe-ballerina-programs/).
+
```
Comment (xlight05) [inline README.md:243]: Did we decide to add combine the major features of Highlights.md with README.md as well? I think we did

File: README.md (hunk 3 above)
Comment (mohanvive) [reply README.md:4]: +1. Will do.

File: README.md (hunk 2 above)
Comment (mohanvive) [reply README.md:244]: Added.

File: productcatalogservice/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Product Catalog Service
+
+The Catalog Service maintains a list of products available in the store. The product list will be read from JSON and converted to a readonly array of `Product` when the service is initialized. The `Catalog Service` has the Search Product capability. This feature is implemented using ballerina query expressions. It allows you to write SQL like queries to filter data from the array. You can find more details about query expressions in this [blog](https://dzone.com/articles/language-integrated-queries-in-ballerina).
+
+```
```
Comment (dilanSachi) [inline productcatalogservice/README.md:5]: `bal` is missing

File: paymentservice/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# Payment Service
+
+Payment Service is responsible for validating the card details and sending a mock payment id. The validation is done by checking length, performing Luhn algorithm validation, validating the card company, and checking the expiry date. This microservice shows some usage of low-level operations to implement the validation algorithm.
+
+
+```bal
+isolated function isLuhnValid() returns boolean|error {
```
Comment (dilanSachi) [inline paymentservice/README.md:7]: After the code improvements, we have missed to update the codes in the readme. These are the old implementations

Comment (mohanvive) [reply paymentservice/README.md:7]: yes, I'll check these places and update the latest code.

File: README.md (hunk 4 above)
Comment (mohanvive) [reply README.md:243]: yes, I am deriving some info to add in the readme. Will update.

File: README.md [hunk 7]
```diff
@@ -1,26 +1,28 @@
...
-| ShippingService | Gives the shipping cost estimates based on the shopping cart. Returns a tracking ID. (Mock)
-| EmailService | Sends the user an order confirmation email with the cart details using the Gmail connector. (mock).
-| CheckoutService | Retrieves the user cart, prepares the order, and orchestrates the payment, shipping, and email notification.
-| RecommendationService | Recommends other products based on the items added to the user’s cart.
-| AdService | Provides text advertisements based on the context of the given words.
+| Service name                                   | Description |
+|------------------------------------------------|-------------|
+| [Frontend](frontend)                           | Exposes an HTTP server to outside to serve data required for the React app. Acts as a frontend for all the backend microservices and abstracts the functionality.|
+| [CartService](cartservice)                     | Stores the product items added to the cart and retrieves them. In memory store and Redis is supported as storage option.
```
Comment (anupama-pathirage) [inline README.md:17]:
```suggestion
| [CartService](cartservice)                     | Stores the product items added to the cart and retrieves them. In memory store and Redis is supported as storage options.
```

File: README.md (hunk 4 above)
Comment (mohanvive) [reply README.md:243]: Done.

File: paymentservice/README.md (hunk 6 above)
Comment (mohanvive) [reply paymentservice/README.md:7]: I have observed that we have avoided using `stubs:` in these examples. Is this international? or these source code added before we these stubs are implemented.

Comment (dilanSachi) [reply paymentservice/README.md:7]: Acually these were added before moving the stub files to a separate module.

**Review by anupama-pathirage — APPROVED** [review]

File: paymentservice/README.md (hunk 6 above)
Comment (mohanvive) [reply paymentservice/README.md:7]: OK, then I'll update those code snippets in the readme files and send another PR.

**Pull request merged**